system/modules/core/classes/Frontend.php

```diff
@@ -469,7 +469,7 @@ protected function jumpToOrReload($intId, $strParams=null, $strForceLang=null)
 			{
 				if ($intId['id'] != $objPage->id  || $blnForceRedirect)
 				{
-					$this->redirect($this->generateFrontendUrl($intId, $strParams, $strForceLang));
```

Comment:
it would potentially alter behavior, because it could already be set. We're doing the same in `PageModel::getFrontendUrl()`